search/result.hpp

```diff
@@ -93,6 +93,9 @@ class Result
   // Precondition: GetResultType() == Type::Feature.
   std::string GetLocalizedFeatureType() const;
   
+  // Precondition: GetResultType() == Type::Feature.
+  std::string GetLocalizedMainFeatureType() const;
```

Comment:
I'd prefer to remove (incapsulate) details like GetLocalizedFeatureType and GetLocalizedMainFeatureType, because suppose that client uses only GetFeatureDescription.

cpp:
```
std::string GetLocalized(uint32_t type) { ... }
...
GetLocalized(m_mainType);
GetLocalized(m_matchedType);
```

Reply:
`GetLocalizedFeatureType` is still needed because it is called from outside. I will remove `GetLocalizedMainFeatureType`.

Reply:
GetLocalizedFeatureType usage is kinda strange.
We have in client something like this now:
```
s = result.GetString();
if (s.empty())
  s = result.GetLocalizedFeatureType();
```
Should be also incapsulated in a function like search::Result::GetTitle()
Maybe next time, ok ..